bccsp/pkcs11/pkcs11.go

```diff
@@ -84,25 +74,44 @@ func loadLib(lib, pin, label string) (*pkcs11.Ctx, uint, *pkcs11.SessionHandle,
 func (csp *impl) getSession() (session pkcs11.SessionHandle) {
 	select {
 	case session = <-csp.sessions:
-		logger.Debugf("Reusing existing pkcs11 session %+v on slot %d\n", session, csp.slot)
+		_, err := csp.ctx.GetSessionInfo(session)
+		if err != nil {
+			logger.Warningf("Get session info failed [%s], closing existing session and getting a new session\n", err)
+			csp.ctx.CloseSession(session)
+			session = createSession(csp.ctx, csp.slot, csp.pin)
```

Comment:
w/o doing some major surgery, this looks ok to me.  the pool should eventually end up with valid sessions.
probably would be good if we could actually recreate the entire ctx, but that's a redesign.